scipy/sparse/tests/test_construct.py

```diff
@@ -373,6 +375,20 @@ def test_bmat(self):
                             r'Got blocks\[0,1\]\.shape\[0\] == 1, expected 2',
                             construct.bmat, [[A,C]])
 
+    @pytest.mark.slow
+    def test_concatenate_int32_overflow():
+        """ test for indptr overflow when concatenating matrices """
+        check_free_memory(30000)
+        
+        n = 33000
+        A = csr_matrix(np.ones((n, n)), dtype=bool))
+        B = A.copy()
+        C = construct._compressed_sparse_stack((A,B), 0)
+        
+        assert_(np.all(np.equal(np.diff(C.indptr), n)))
+        assert_equal(C.indices.dtype, np.int64)
+        assert_equal(C.indptr.dtype, np.int64)
+
```

Comment:
Another test that makes sure we aren't promoting to int64 if we don't need to would be great (and won't require huge RAM to test).